src/libraries/System.IO.Compression/tests/ZipArchive/zip_ReadTests.cs

```diff
@@ -538,15 +578,177 @@ public static async Task ReadStreamOps(bool async)
+            using (var ms = new MemoryStream())
+            {
+                // Create a ZIP with stored entries
+                using (var archive = new ZipArchive(ms, ZipArchiveMode.Create, true))
+                {
+                    var entry = archive.CreateEntry("test.txt", CompressionLevel.NoCompression);
+                    using (var stream = entry.Open())
+                    {
+                        var testData = "This is test data for seeking operations."u8.ToArray();
+                        stream.Write(testData, 0, testData.Length);
+                    }
+                }
+
+                ms.Position = 0;
+                using (var archive = await CreateZipArchive(async, ms, ZipArchiveMode.Read))
+                {
+                    foreach (ZipArchiveEntry e in archive.Entries)
+                    {
+                        if (e.Length == 0) continue; // Skip empty entries for this test
+
+                        Stream s = await OpenEntryStream(async, e);
+
+                        // For stored entries, SubReadStream should be seekable when underlying stream is seekable
+                        Assert.True(s.CanSeek, $"SubReadStream should be seekable for stored entry '{e.FullName}' when underlying stream is seekable");
+
+                        // Test seeking to beginning
+                        long pos = s.Seek(0, SeekOrigin.Begin);
+                        Assert.Equal(0, pos);
+                        Assert.Equal(0, s.Position);
+
+                        // Test seeking to end
+                        pos = s.Seek(0, SeekOrigin.End);
+                        Assert.Equal(e.Length, pos);
+                        Assert.Equal(e.Length, s.Position);
+
+                        // Test seeking from current position
+                        s.Position = 0;
+                        if (e.Length > 1)
+                        {
+                            pos = s.Seek(1, SeekOrigin.Current);
+                            Assert.Equal(1, pos);
+                            Assert.Equal(1, s.Position);
+                        }
+
+                        // Test setting position directly
+                        s.Position = 0;
+                        Assert.Equal(0, s.Position);
+
+                        // Test that seeking beyond bounds throws
+                        Assert.Throws<ArgumentOutOfRangeException>(() => s.Position = -1);
+                        Assert.Throws<ArgumentOutOfRangeException>(() => s.Position = e.Length + 1);
+                        Assert.Throws<IOException>(() => s.Seek(-1, SeekOrigin.Begin));
+                        Assert.Throws<IOException>(() => s.Seek(1, SeekOrigin.End));
+
+                        await DisposeStream(async, s);
+                    }
+                }
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(Get_Booleans_Data))]
+        public static async Task ReadEntryContentTwice(bool async)
```

Comment:
rewrite this test to accept additional boolean parameter that would specify whether the Seek method or the Position setter should be used to rewind back to the beginning of the entry.